### PR TITLE
Share common structs in common/v1 and adds missing fields to v2alpha1

### PR DESCRIPTION
**Pull request by vboulineau**
### What does this PR do?

Add some missing fields to `v2alpha1` to allow conversion from `v1alpha1`.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

No test required.

### CODE REVIEW

File: apis/datadoghq/v2alpha1/datadogagent_types.go [hunk 1]
```diff
@@ -228,6 +266,40 @@ type USMFeatureConfig struct {
...
+	// Can be passed as raw data or via a json encoded string in a config map.
+	// See also: https://docs.datadoghq.com/developers/dogstatsd/dogstatsd_mapper/
+	// +optional
+	MapperProfiles *CustomConfig `json:"mapperProfiles,omitempty"`
+}
+
+// EventCollectionConfig contains the Event Collection configuration.
+// +k8s:openapi-gen=true
+type EventCollectionConfig struct {
+	KubernetesEventsCollection *bool `json:"kubernetesEventsCollection,omitempty"`
```
Comment (celenechang) [inline apis/datadoghq/v2alpha1/datadogagent_types.go:300]:
```suggestion
	CollectKubernetesEvents *bool `json:"collectKubernetesEvents,omitempty"`
```
suggesting an "active" format to be consistent with other `bool`s. 
I assume the idea is to support additional event types in the future?

File: docs/configuration.md [hunk 2]
```diff
@@ -183,10 +183,10 @@ spec:
 | agent.hostNetwork | Host networking requested for this pod. Use the host's network namespace. If this option is set, the ports that will be used must be specified. Default to false. |
 | agent.hostPID | Use the host's pid namespace. Optional: Default to false. |
 | agent.image.jmxEnabled | Define whether the Agent image should support JMX. |
-| agent.image.name | Define the image to use: Use "gcr.io/datadoghq/agent" for Datadog Agent 7 Use "datadog/dogstatsd" for Standalone Datadog Agent DogStatsD Use "gcr.io/datadoghq/cluster-agent" for Datadog Cluster Agent Use "agent" with the registry and tag configurations for <registry>/agent:<tag> Use "cluster-agent" with the registry and tag configurations for <registry>/cluster-agent:<tag> |
+| agent.image.name | Define the image to use: Use "gcr.io/datadoghq/agent:latest" for Datadog Agent 7 Use "datadog/dogstatsd:latest" for standalone Datadog Agent DogStatsD 7 Use "gcr.io/datadoghq/cluster-agent:latest" for Datadog Cluster Agent Use "agent" with the registry and tag configurations for <registry>/agent:<tag> Use "cluster-agent" with the registry and tag configurations for <registry>/cluster-agent:<tag> |
```
Comment (maycmlee) [inline docs/configuration.md:186]:
```suggestion
| agent.image.name | Define the image to use: Use "gcr.io/datadoghq/agent:latest" for Datadog Agent 7. Use "datadog/dogstatsd:latest" for standalone Datadog Agent DogStatsD 7. Use "gcr.io/datadoghq/cluster-agent:latest" for Datadog Cluster Agent. Use "agent" with the registry and tag configurations for <registry>/agent:<tag>. Use "cluster-agent" with the registry and tag configurations for <registry>/cluster-agent:<tag>. |
```

File: docs/configuration.md [hunk 3]
```diff
@@ -309,10 +309,10 @@ spec:
 | clusterAgent.deploymentName | Name of the Cluster Agent Deployment to create or migrate from. |
 | clusterAgent.enabled | Enabled |
 | clusterAgent.image.jmxEnabled | Define whether the Agent image should support JMX. |
-| clusterAgent.image.name | Define the image to use: Use "gcr.io/datadoghq/agent" for Datadog Agent 7 Use "datadog/dogstatsd" for Standalone Datadog Agent DogStatsD Use "gcr.io/datadoghq/cluster-agent" for Datadog Cluster Agent Use "agent" with the registry and tag configurations for <registry>/agent:<tag> Use "cluster-agent" with the registry and tag configurations for <registry>/cluster-agent:<tag> |
+| clusterAgent.image.name | Define the image to use: Use "gcr.io/datadoghq/agent:latest" for Datadog Agent 7 Use "datadog/dogstatsd:latest" for standalone Datadog Agent DogStatsD 7 Use "gcr.io/datadoghq/cluster-agent:latest" for Datadog Cluster Agent Use "agent" with the registry and tag configurations for <registry>/agent:<tag> Use "cluster-agent" with the registry and tag configurations for <registry>/cluster-agent:<tag> |
```
Comment (maycmlee) [inline docs/configuration.md:312]:
```suggestion
| clusterAgent.image.name | Define the image to use: Use "gcr.io/datadoghq/agent:latest" for Datadog Agent 7. Use "datadog/dogstatsd:latest" for standalone Datadog Agent DogStatsD 7. Use "gcr.io/datadoghq/cluster-agent:latest" for Datadog Cluster Agent. Use "agent" with the registry and tag configurations for <registry>/agent:<tag>. Use "cluster-agent" with the registry and tag configurations for <registry>/cluster-agent:<tag>. |
```

File: docs/configuration.md [hunk 4]
```diff
@@ -375,10 +375,10 @@ spec:
 | clusterChecksRunner.deploymentName | Name of the cluster checks deployment to create or migrate from. |
 | clusterChecksRunner.enabled | Enabled |
 | clusterChecksRunner.image.jmxEnabled | Define whether the Agent image should support JMX. |
-| clusterChecksRunner.image.name | Define the image to use: Use "gcr.io/datadoghq/agent" for Datadog Agent 7 Use "datadog/dogstatsd" for Standalone Datadog Agent DogStatsD Use "gcr.io/datadoghq/cluster-agent" for Datadog Cluster Agent Use "agent" with the registry and tag configurations for <registry>/agent:<tag> Use "cluster-agent" with the registry and tag configurations for <registry>/cluster-agent:<tag> |
+| clusterChecksRunner.image.name | Define the image to use: Use "gcr.io/datadoghq/agent:latest" for Datadog Agent 7 Use "datadog/dogstatsd:latest" for standalone Datadog Agent DogStatsD 7 Use "gcr.io/datadoghq/cluster-agent:latest" for Datadog Cluster Agent Use "agent" with the registry and tag configurations for <registry>/agent:<tag> Use "cluster-agent" with the registry and tag configurations for <registry>/cluster-agent:<tag> |
```
Comment (maycmlee) [inline docs/configuration.md:378]:
```suggestion
| clusterChecksRunner.image.name | Define the image to use: Use "gcr.io/datadoghq/agent:latest" for Datadog Agent 7. Use "datadog/dogstatsd:latest" for standalone Datadog Agent DogStatsD 7. Use "gcr.io/datadoghq/cluster-agent:latest" for Datadog Cluster Agent. Use "agent" with the registry and tag configurations for <registry>/agent:<tag>. Use "cluster-agent" with the registry and tag configurations for <registry>/cluster-agent:<tag>. |
```

**Review by maycmlee — APPROVED** [review]
Small edits

File: apis/datadoghq/v2alpha1/datadogagent_types.go [hunk 5]
```diff
@@ -165,10 +170,32 @@ type ProcessCollectionFeatureConfig struct {
...
+	// +optional
+	Kubelet *commonv1.KubeletConfig `json:"kubelet,omitempty"`
+
+	// Path to the docker runtime socket.
+	// +optional
+	DockerSocketPath *string `json:"dockerSocketPath,omitempty"`
+
+	// Path to the container runtime socket (if different from Docker).
+	// +optional
+	CriSocketPath *string `json:"criSocketPath,omitempty"`
```
Comment (celenechang) [inline apis/datadoghq/v2alpha1/datadogagent_types.go:198]: Possibly move to GlobalConfig?

File: apis/datadoghq/v2alpha1/datadogagent_types.go [hunk 6]
```diff
@@ -165,10 +170,32 @@ type ProcessCollectionFeatureConfig struct {
...
+
+	// Enables the OOMKill eBPF-based check.
+	// Default: false
+	// +optional
+	OOMKillCheck *bool `json:"oomKillCheck,omitempty"`
+
+	// Enables the TCP queue length eBPF-based check.
+	// Default: false
+	// +optional
+	TCPQueueLengthCheck *bool `json:"tcpQueueLengthCheck,omitempty"`
```
Comment (celenechang) [inline apis/datadoghq/v2alpha1/datadogagent_types.go:186]: add dedicated structs

File: apis/datadoghq/v2alpha1/datadogagent_types.go [hunk 7]
```diff
@@ -48,6 +49,10 @@ type DatadogFeatures struct {
 	ProcessCollection *ProcessCollectionFeatureConfig `json:"processCollection,omitempty"`
 	// ContainerCollection configuration.
 	ContainerCollection *ContainerCollectionFeatureConfig `json:"containerCollection,omitempty"`
```
Comment (celenechang) [inline apis/datadoghq/v2alpha1/datadogagent_types.go:51]: Can we change this to LiveContainerCollection?

Comment (vboulineau) [reply apis/datadoghq/v2alpha1/datadogagent_types.go:51]: `LiveProcessCollection` as well?

Comment (vboulineau) [reply apis/datadoghq/v2alpha1/datadogagent_types.go]: Implemented both

File: apis/datadoghq/common/v1/agent_types.go [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+// AgentImageConfig defines the agent container image config.
+// +kubebuilder:object:generate=true
+type AgentImageConfig struct {
+	// Define the image to use:
+	// Use "gcr.io/datadoghq/agent:latest" for Datadog Agent 7.
+	// Use "datadog/dogstatsd:latest" for standalone Datadog Agent DogStatsD 7.
+	// Use "gcr.io/datadoghq/cluster-agent:latest" for Datadog Cluster Agent.
+	// Use "agent" with the registry and tag configurations for <registry>/agent:<tag>.
+	// Use "cluster-agent" with the registry and tag configurations for <registry>/cluster-agent:<tag>.
+	Name string `json:"name,omitempty"`
```
Comment (celenechang) [inline apis/datadoghq/common/v1/agent_types.go:19]: i'm a little confused by the comments; should we have a `Registry` field also?

File: apis/datadoghq/v2alpha1/datadogagent_types.go [hunk 9]
```diff
@@ -6,20 +6,21 @@
...
 
-// ResourceName is the name of a Deployment Component
-type ResourceName string
+// ComponentName is the name of a Deployment Component
+type ComponentName string
 
 const (
 	// NodeAgentResourceName is the name of the Datadog Node Agent
-	NodeAgentResourceName ResourceName = "nodeAgent"
+	NodeAgentResourceName ComponentName = "nodeAgent"
```
Comment (celenechang) [inline apis/datadoghq/v2alpha1/datadogagent_types.go:19]:
```suggestion
	NodeAgentComponentName ComponentName = "nodeAgent"
```

File: apis/datadoghq/v2alpha1/datadogagent_types.go [hunk 10]
```diff
@@ -44,10 +45,14 @@ type DatadogFeatures struct {
 
 	// LogCollection configuration.
 	LogCollection *LogCollectionFeatureConfig `json:"logCollection,omitempty"`
-	// ProcessCollection configuration.
-	ProcessCollection *ProcessCollectionFeatureConfig `json:"processCollection,omitempty"`
+	// LiveProcessCollection configuration.
+	LiveProcessCollection *LiveProcessCollectionFeatureConfig `json:"liveProcessCollection,omitempty"`
 	// ContainerCollection configuration.
```
Comment (celenechang) [inline apis/datadoghq/v2alpha1/datadogagent_types.go:50]:
```suggestion
	// LiveContainerCollection configuration.
```

File: apis/datadoghq/v2alpha1/datadogagent_types.go [hunk 11]
```diff
@@ -44,10 +45,14 @@ type DatadogFeatures struct {
...
-	ContainerCollection *ContainerCollectionFeatureConfig `json:"containerCollection,omitempty"`
+	LiveContainerCollection *LiveContainerCollectionFeatureConfig `json:"liveContainerCollection,omitempty"`
+	// ContainerCollection configuration.
```
Comment (celenechang) [inline apis/datadoghq/v2alpha1/datadogagent_types.go:52]:
```suggestion
	// OOMKill configuration.
```

File: apis/datadoghq/v2alpha1/datadogagent_types.go [hunk 12]
```diff
@@ -44,10 +45,14 @@ type DatadogFeatures struct {
...
+	OOMKill *OOMKillFeatureConfig `json:"oomKill,omitempty"`
+	// ContainerCollection configuration.
```
Comment (celenechang) [inline apis/datadoghq/v2alpha1/datadogagent_types.go:54]:
```suggestion
	// TCPQueueLength configuration.
```

File: apis/datadoghq/v2alpha1/datadogagent_types.go [hunk 13]
```diff
@@ -198,16 +223,16 @@ type CWSFeatureConfig struct {
 	// +optional
 	Enabled *bool `json:"enabled,omitempty"`
 
+	// EnableSyscallMonitor enables Syscall Monitoring (recommended for troubleshooting only).
+	// Default: false
+	// +optional
+	EnableSyscallMonitor *bool `json:"enableSyscallMonitor,omitempty"`
```
Comment (celenechang) [inline apis/datadoghq/v2alpha1/datadogagent_types.go:229]: i wonder if we should have passive voice to be parallel with the `Enabled` parameters; i.e. `SyscallMonitorEnabled`

File: apis/datadoghq/v2alpha1/datadogagent_types.go [hunk 14]
```diff
@@ -228,6 +264,40 @@ type USMFeatureConfig struct {
 	Enabled *bool `json:"enabled,omitempty"`
 }
 
+// DogstatsdConfig contains the Dogstatsd configuration parameters.
+// +k8s:openapi-gen=true
+type DogstatsdConfig struct {
+	// Enable origin detection for container tagging.
+	// See also: https://docs.datadoghq.com/developers/dogstatsd/unix_socket/#using-origin-detection-for-container-tagging
+	// +optional
+	OriginDetection *bool `json:"originDetection,omitempty"`
```
Comment (celenechang) [inline apis/datadoghq/v2alpha1/datadogagent_types.go:273]:
```suggestion
	OriginDetectionEnabled *bool `json:"originDetectionEnabled,omitempty"`
```
?

File: apis/datadoghq/v2alpha1/datadogagent_types.go [hunk 15]
```diff
@@ -545,34 +627,98 @@ type LocalService struct {
...
+	// +optional
 	Name string `json:"name,omitempty"`
-}
 
-// DatadogAgentPodTemplateOverride is the generic description equivalent to a subset of the PodTemplate for a component.
-type DatadogAgentPodTemplateOverride struct {
-	// Configure the basic configurations for each agent container
-	Containers []DatadogAgentGenericContainer `json:"containers,omitempty"`
+	// Number of the replicas.
+	// Not applicable for DaemonSet/ExtendedDaemonSet
```
Comment (celenechang) [inline apis/datadoghq/v2alpha1/datadogagent_types.go:671]:
```suggestion
	// Not applicable for a DaemonSet/ExtendedDaemonSet deployment
```

File: apis/datadoghq/v2alpha1/datadogagent_types.go [hunk 16]
```diff
@@ -588,47 +734,57 @@ type DatadogAgentPodTemplateOverride struct {
...
 	// +optional
-	Host *corev1.EnvVarSource `json:"host,omitempty"`
+	HostPID *bool `json:"hostPID,omitempty"`
 
-	// TLSVerify toggles kubelet TLS verification.
-	// Default: true
+	// SecCompRootPath specify the seccomp profile root directory.
 	// +optional
-	TLSVerify *bool `json:"tlsVerify,omitempty"`
+	SecCompRootPath string `json:"secCompRootPath,omitempty"`
```
Comment (celenechang) [inline apis/datadoghq/v2alpha1/datadogagent_types.go:764]: should it be `*string` ?

File: apis/datadoghq/v2alpha1/datadogagent_types.go [hunk 17]
```diff
@@ -588,47 +734,57 @@ type DatadogAgentPodTemplateOverride struct {
...
 	// +optional
-	HostCAPath string `json:"hostCAPath,omitempty"`
+	SecCompCustomProfile *CustomConfig `json:"secCompCustomProfile,omitempty"`
 
-	// AgentCAPath is the container path where the kubelet CA certificate is stored.
-	// Default: '/var/run/host-kubelet-ca.crt' if hostCAPath is set, else '/var/run/secrets/kubernetes.io/serviceaccount/ca.crt'
+	// SecCompProfileName specify a seccomp profile.
 	// +optional
-	AgentCAPath string `json:"agentCAPath,omitempty"`
+	SecCompProfileName string `json:"secCompProfileName,omitempty"`
```
Comment (celenechang) [inline apis/datadoghq/v2alpha1/datadogagent_types.go:772]: same, perhaps `*string` ?

File: apis/datadoghq/common/v1/types.go [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package common
+
+import corev1 "k8s.io/api/core/v1"
+
+// Secret contains a secret name and an included key.
+// +kubebuilder:object:generate=true
+type Secret struct {
```
Comment (celenechang) [inline apis/datadoghq/common/v1/types.go:12]: should we call it `SecretConfig` for consistency?

File: apis/datadoghq/common/v1/agent_types.go (hunk 8 above)
Comment (vboulineau) [reply apis/datadoghq/common/v1/agent_types.go:19]: Registry is in `GlobalConfig`

**Review by celenechang — APPROVED** [review]

**Pull request merged**